torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -121,6 +121,18 @@ RegisterOperators reg({
             return 0;
           };
         }),
+    Operator(
+        "prim::range(int n) -> int[]",
```

Comment:
Hmm I'm a bit ambivalent about adding this, since those objects have completely different characteristics in Python.
```
for i in range(2 ** 64):
    ...
```
is fine in Python 3, but we're using Python 2 semantics in TorchScript? That seems like not a bad choice.

Reply:
Yea I wasn't sure about this but it seems very useful atm as it saves me from writing for unnecessary loops in torchscript. 
An alternative of this I thought about was list comprehension which in general I think it's also super useful, I went for prim::range because it's  quick to implement I think. Possibly we could not document this feature and remove it when list comprehension is ready?

Reply:
I'm fine with adding a special primitive, just don't call it `range`. If we use Python names we should match Python semantics.

Reply:
Yea that makes sense, I will rename it in a following PR. :) Thanks!